[Dexter Kwan] iP

dexterkwxn opened the pull request:
### Duchess

> _“Your mind is for having ideas, not holding them.”_ – David Allen [(source)](https://gettingthingsdone.com/)

Duchess helps you stay organized by freeing your mind from remembering tasks. It's:

- **Text-based**
- **Easy to learn**
- **~~Fast~~ SUPER FAST** to use

#### How to Use
1. [Download it from here](#).
2. Double-click to open.
3. Add your tasks.
4. Let it manage your tasks for you! 😉  

And the best part? **It's FREE!**  

#### Features:
- [x] Managing tasks  
- [ ] Managing deadlines (coming soon)  
- [ ] Reminders (coming soon)  

If you're a Java programmer, you can use it to practice Java too. Here's the `main` method:

```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

dexterkwxn committed:
Add greet and exit message

dexterkwxn committed:
Add echo

dexterkwxn committed:
add List

dexterkwxn committed:
Add marking of items

dexterkwxn committed:
Add Todo, Deadlines, Events

dexterkwxn committed:
Add automated testing

dexterkwxn committed:
Add exceptions

dexterkwxn committed:
Add delete function

dexterkwxn committed:
Add save and load list function

dexterkwxn committed:
Merge branch 'branch-level-7'

dexterkwxn committed:
Add support for DateTime

dexterkwxn committed:
Merge branch 'branch-level-8'

dexterkwxn committed:
Add more OOP

dexterkwxn committed:
Moved files into packages

dexterkwxn committed:
Merge branch 'add-gradle-support'

dexterkwxn committed:
Moved data directory

dexterkwxn committed:
Add gradle support

dexterkwxn committed:
Add TaskListTest and TodoTest

dexterkwxn committed:
Add shadowJar config

dexterkwxn committed:
Add JavaDoc comments

dexterkwxn committed:
Add find feature

dexterkwxn committed:
Refactor code to follow coding standard

dexterkwxn committed:
Merge branch 'branch-A-JavaDoc'

dexterkwxn committed:
Merge branch 'branch-A-CodingStandard'

dexterkwxn committed:
Merge branch 'branch-Level-9'

Luoq1-Xu commented on src/main/java/duchess/Duchess.java:
```diff
@@ -0,0 +1,226 @@
+        while (isRunning) {
+            try {
+                in = this.ui.readCommand();
+                String[] commandStr= this.parser.parseCommand(in);
```
May have missed a space here.

```suggestion
                String[] commandStr = this.parser.parseCommand(in);
```

Luoq1-Xu commented on src/main/java/duchess/Duchess.java:
```diff
@@ -0,0 +1,226 @@
+                        break;
+                    default:
+                        this.processUnrecognisedCommand(in);
+                        break;
```
There should be no indentation for `case` clauses, according to the Java Coding Standards for the course. Do consider editing this chunk

Luoq1-Xu commented on src/main/java/duchess/Duchess.java:
```diff
@@ -0,0 +1,226 @@
+        } catch (Exception e) {
+            throw new DuchessException(in, ErrorType.INVALID_FORMAT);
+        }
+    }
```
Would probably be good to add a check here in case the user specifies an invalid index or there are no tasks left. User might be left confused if they entered a command with the right format but get an Invalid Format error message.

Luoq1-Xu commented on src/main/java/duchess/DuchessException.java:
```diff
@@ -0,0 +1,31 @@
+        String s;
+        switch(this.errorType) {
+            case INVALID_COMMAND:
+                s = "What are you blabbering about you babboon.";
```
Nice error messages 😆 
Quite informative

Luoq1-Xu commented on src/main/java/duchess/Storage.java:
```diff
@@ -0,0 +1,104 @@
+     *
+     * @return A TaskList containing tasks loaded from the file.
+     */
+    public TaskList loadList() {
```
The indentation of the javaDocs is not aligned to the method signature here

Luoq1-Xu commented on src/main/java/duchess/Storage.java:
```diff
@@ -0,0 +1,104 @@
+        return taskList;
+    }
+
+     /**
```
Same thing here, javaDocs is not aligned

Luoq1-Xu reviewed (commented):
Overall the code is well organised and quite well documented. I added some slight syntax suggestions, and consider splitting some logic from the main `duchess.java` class into other classes.

dexterkwxn committed:
Add OOB detection for deleteTask()

keyansheng commented on src/main/java/duchess/Deadline.java:
```diff
@@ -0,0 +1,48 @@
+ */
+public class Deadline extends Task{
+    LocalDateTime by;
+    public final static String taskSymbol = "D";
```
It will be good to use SCREAMING_SNAKE_CASE for constants.

keyansheng commented on src/main/java/duchess/Deadline.java:
```diff
@@ -0,0 +1,48 @@
+ * A {@code Deadline} object stores the task name and the due date/time.
+ */
+public class Deadline extends Task{
+    LocalDateTime by;
```
`by` can be set to private since it's not used anywhere else and can be kept encapsulated.
```suggestion
    private LocalDateTime by;
```

keyansheng replied:
Should instance variables be placed above or below static variables?

keyansheng commented on src/main/java/duchess/Deadline.java:
```diff
@@ -0,0 +1,48 @@
+ * Represents a task with a deadline. 
+ * A {@code Deadline} object stores the task name and the due date/time.
+ */
+public class Deadline extends Task{
```
A space can be added here for consistency and readability.
```suggestion
public class Deadline extends Task {
```

keyansheng commented on src/main/java/duchess/Duchess.java:
```diff
@@ -0,0 +1,229 @@
+     * @param in The user input containing the task description.
+     * @throws DuchessException If the input format is invalid.
+     */
+    public void addTodo(String in) throws DuchessException  {
```
There is an extra space here.
```suggestion
    public void addTodo(String in) throws DuchessException {
```

keyansheng commented on src/main/java/duchess/Duchess.java:
```diff
@@ -0,0 +1,229 @@
+            throw new DuchessException(in, ErrorType.INVALID_FORMAT);
+        }
+    }
+    /**
```
Consider adding a blank line between methods for consistency and clearer separation.

keyansheng reviewed (commented):
Looks good overall, with only minor formatting improvements.

dexterkwxn committed:
Add checkStyle

dexterkwxn committed:
Merge branch 'branch-A-checkStyle'

dexterkwxn committed:
Add JavaFX UI support

dexterkwxn committed:
Merge branch 'branch-Level-10'

dexterkwxn committed:
Add Assertion in Duchess.java

Add assertion to processInput() method in Duchess.java

Ensures that the response message sent to the application is not empty
after all is said and done.

dexterkwxn committed:
Refactoring Task Classes

Improve code quality in Task classes

Changed static final string from taskSymbol to TASK_SYMBOL for better
readability.

dexterkwxn committed:
Merge pull request #1 from dexterkwxn/branch-A-Assertions

Add Assertion in Duchess.java

dexterkwxn committed:
Merge branch 'master' into branch-A-CodeQuality

dexterkwxn committed:
Remove unused code in Duchess.java

dexterkwxn committed:
Merge pull request #2 from dexterkwxn/branch-A-CodeQuality

Merge Code Quality Branch

dexterkwxn committed:
Add sorting for Task List

dexterkwxn committed:
Modify error handling for input

dexterkwxn committed:
Fix checkstyle warnings

dexterkwxn committed:
Add README.md
